src/implementations/vecs.rs

```diff
@@ -2,59 +2,309 @@ use crate::DeserializeRevisioned;
+macro_rules! impl_revisioned_specialised_vec {
+	($ty:ty) => {
+		impl SerializeRevisioned for Vec<$ty> {
+			#[inline]
+			fn serialize_revisioned<W: Write>(&self, writer: &mut W) -> Result<(), Error> {
+				// Write the length first (number of elements)
+				self.len().serialize_revisioned(writer)?;
+				// For zero-length vectors, return early
+				if self.is_empty() {
+					return Ok(());
+				}
+				// On little-endian platforms, numbers are already in the correct byte
+				// order, whilst on big-endian platforms, we need to convert them.
+				if cfg!(target_endian = "little") {
+					// This is safe because:
+					// 1. This type has a well-defined byte representation
+					// 2. On little-endian platforms, memory representation matches wire format
+					// 3. We're only reading from the slice, not modifying it
+					unsafe {
+						let byte_slice = std::slice::from_raw_parts(
+							self.as_ptr().cast::<u8>(),
+							self.len() * std::mem::size_of::<$ty>(),
+						);
+						writer.write_all(byte_slice).map_err(Error::Io)
+					}
+				} else {
+					// Slow path: per-element little-endian conversion
+					for value in self.iter() {
+						writer.write_all(&value.to_le_bytes()).map_err(Error::Io)?;
+					}
+					Ok(())
+				}
+			}
+		}
+
+		impl DeserializeRevisioned for Vec<$ty> {
+			#[inline]
+			fn deserialize_revisioned<R: Read>(reader: &mut R) -> Result<Self, Error> {
+				// Read the length first (number of elements)
+				let len = usize::deserialize_revisioned(reader)?;
+				// For zero-length vectors, return early
+				if len == 0 {
+					return Ok(Self::new());
+				}
+				// On little-endian platforms, numbers are already in the correct byte
+				// order, whilst on big-endian platforms, we need to convert them.
+				if cfg!(target_endian = "little") {
+					// Fast path: bulk read directly into Vec
+					let byte_len = len
+						.checked_mul(std::mem::size_of::<$ty>())
+						.ok_or(Error::IntegerOverflow)?;
+					// Allocate initialized buffer to ensure proper alignment and safety
+					let mut vec = vec![<$ty>::default(); len];
+					// Read the bytes into the vector
+					unsafe {
+						let byte_slice =
+							std::slice::from_raw_parts_mut(vec.as_mut_ptr().cast::<u8>(), byte_len);
+						reader.read_exact(byte_slice).map_err(Error::Io)?;
+					}
+					// Return the vector
+					Ok(vec)
+				} else {
+					// Create a vector with the necessary capacity
+					let mut vec = Vec::with_capacity(len);
+					// Slow path: per-element little-endian conversion
+					for _ in 0..len {
+						// Read the bytes into a temporary buffer
+						let mut b = [0u8; std::mem::size_of::<$ty>()];
+						reader.read_exact(&mut b).map_err(Error::Io)?;
+						// Convert the bytes to the target type
+						let v = <$ty>::from_le_bytes(b);
+						// Check if the vector is at capacity
+						if vec.len() > vec.capacity() {
+							unsafe { std::hint::unreachable_unchecked() }
+						}
```

Comment:
It might actually be better to use the more idiomatic function for this:
```suggestion
						unsafe{ std::hint::assert_unchecked(vec.len() < vec.capacity()) }
```

Reply:
Fixed in [1a9d07e](https://github.com/surrealdb/revision/pull/43/commits/1a9d07e58ea0517e7bf24a1a74beebe2b316c389)